src/rand_tangent.jl

```diff
@@ -33,7 +33,12 @@ function rand_tangent(rng::AbstractRNG, x::T) where {T}
         tangents = map(field_names) do field_name
             rand_tangent(rng, getfield(x, field_name))
         end
-        return Composite{T}(; NamedTuple{field_names}(tangents)...)
+        if all(tangent isa DoesNotExist for tangent in tangents)
+            # if none of my fields can be perturbed then I can't be perturbed
+            return DoesNotExist()
+        else
+            Composite{T}(; NamedTuple{field_names}(tangents)...)
+        end
     else
         return NO_FIELDS
```

Comment:
The outer `if` here that gives us the `NO_FIELDS` would be able to be removed if we replaced `NO_FIELDS` with `DoesNotExist()`.
Since then we would be in the case that all the fields (of which there are none) subtype `DoesNotExist`.
Vacuous truth.

https://github.com/JuliaDiff/ChainRulesCore.jl/issues/175#issuecomment-829632604